mlrun/api/initial_data.py

```diff
@@ -508,33 +539,147 @@ def _rename_marketplace_kind_to_hub(
         db._upsert(db_session, [hub], ignore=True)
 
 
-def _perform_version_1_data_migrations(
+def _perform_version_4_data_migrations(
     db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
-    _enrich_project_state(db, db_session)
-    _fix_artifact_tags_duplications(db, db_session)
-    _fix_datasets_large_previews(db, db_session)
+    _migrate_artifacts_table_v2(db, db_session)
 
 
-def _enrich_project_state(
+def _migrate_artifacts_table_v2(
     db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
-    logger.info("Enriching projects state")
-    projects = db.list_projects(db_session)
-    for project in projects.projects:
-        changed = False
-        if not project.spec.desired_state:
-            changed = True
-            project.spec.desired_state = mlrun.common.schemas.ProjectState.online
-        if not project.status.state:
-            changed = True
-            project.status.state = project.spec.desired_state
-        if changed:
-            logger.debug(
-                "Found project without state data. Enriching",
-                name=project.metadata.name,
+    """
+    Migrate the old artifacts table to the new one, which contains the following columns:
+    uid, key, project, iteration, kind, producer_id, created, updated, and full_object
+
+    Also migrate the artifact labels and tags to the new tables
+    """
+    logger.info("Aligning artifacts")
+    new_artifacts = []
+    artifacts_tags_to_migrate = []
+    artifacts_labels_to_migrate = []
+    artifacts = db._find_artifacts(db_session, None, "*")
+    for artifact in artifacts:
+        new_artifact = mlrun.api.db.sqldb.models.ArtifactV2()
+        artifact_dict = artifact.struct
+        artifact_metadata = {}
+
+        # if it is a legacy artifact, it won't a "metadata" field and the artifact_dict will contain the metadata
+        legacy = is_legacy_artifact(artifact_dict)
```

Comment:
You should convert the object if it's legacy. This will make the code easier to read, and will also make sure we don't end up with legacy objects in the DB (which is part of the point here).